src/Common/src/Interop/OSX/System.Security.Cryptography.Native.Apple/Interop.SecKeyRef.Export.cs

```diff
@@ -0,0 +1,202 @@
+        private static DerSequenceReader ReadEncryptedPkcs8Blob(string passphrase, DerSequenceReader reader)
+        {
+            // EncryptedPrivateKeyInfo::= SEQUENCE {
+            //    encryptionAlgorithm EncryptionAlgorithmIdentifier,
+            //    encryptedData        EncryptedData }
+            //
+            // EncryptionAlgorithmIdentifier ::= AlgorithmIdentifier
+            //
+            // EncryptedData ::= OCTET STRING
+            DerSequenceReader algorithmIdentifier = reader.ReadSequence();
+            string algorithmOid = algorithmIdentifier.ReadOidAsString();
+
+            // PBES2 (Password-Based Encryption Scheme 2)
+            if (algorithmOid != OidPbes2)
+            {
+                throw new CryptographicException();
+            }
+
+            // PBES2-params ::= SEQUENCE {
+            //    keyDerivationFunc AlgorithmIdentifier { { PBES2 - KDFs} },
+            //    encryptionScheme AlgorithmIdentifier { { PBES2 - Encs} }
+            // }
+
+            DerSequenceReader pbes2Params = algorithmIdentifier.ReadSequence();
+            algorithmIdentifier = pbes2Params.ReadSequence();
+
+            string kdfOid = algorithmIdentifier.ReadOidAsString();
+
+            // PBKDF2 (Password-Based Key Derivation Function 2)
+            if (kdfOid != OidPbkdf2)
+            {
+                throw new CryptographicException();
+            }
+
+            // PBKDF2-params ::= SEQUENCE {
+            //   salt CHOICE {
+            //     specified OCTET STRING,
+            //     otherSource AlgorithmIdentifier { { PBKDF2 - SaltSources} }
+            //   },
+            //   iterationCount INTEGER (1..MAX),
+            //   keyLength INTEGER(1..MAX) OPTIONAL,
+            //   prf AlgorithmIdentifier { { PBKDF2 - PRFs} }  DEFAULT algid - hmacWithSHA1
+            // }
+            DerSequenceReader pbkdf2Params = algorithmIdentifier.ReadSequence();
+
+            byte[] salt = pbkdf2Params.ReadOctetString();
+            int iterCount = pbkdf2Params.ReadInteger();
+            int keySize = -1;
+
+            if (pbkdf2Params.HasData && pbkdf2Params.PeekTag() == (byte)DerSequenceReader.DerTag.Integer)
+            {
+                keySize = pbkdf2Params.ReadInteger();
+            }
+
+            if (pbkdf2Params.HasData)
+            {
+                string prfOid = pbkdf2Params.ReadOidAsString();
+
+                // SHA-1 is the only hash algorithm our PBKDF2 supports.
+                if (prfOid != OidSha1)
+                {
+                    throw new CryptographicException(prfOid);
+                }
+            }
+
+            DerSequenceReader encryptionScheme = pbes2Params.ReadSequence();
+            string cipherOid = encryptionScheme.ReadOidAsString();
+
+            // DES-EDE3-CBC (TripleDES in CBC mode)
+            if (cipherOid != OidTripleDesCbc)
+            {
+                throw new CryptographicException();
```

Comment:
Good call.  I added them all as Debug.Fail messages.  They felt a bit too inner-state to be exception text.  There's nothing a user can do about it, it means the platform changed something, not them.